### ISSUE TITLE
Allow configurable subdomains

### ISSUE THREAD

**Issue by animeai** [davestephens/ansible-nas#109]
ansible-nas/templates/traefik/traefik.toml lines 181 onwards.

Currently traefik will be requesting certificates for all subdomains whether we require them or not. This is not good practice as we will be hitting the letsencrypt daily limits when initially setting up. It would be better to use an ansible task to create the traefik.toml file based upon the services loaded. This is a little beyond my ansible expertise though! Currently I can only think that a series of plays would be required to do it building onto a variable string if a task is enabled before using a sed command to merge into the traefik.toml file... Something like: (shameless copy/paste of code form elsewhere to save typing!)

```
        - name: Initialize an empty list for our strings
	  set_fact:
	    my_strings: []

	- name: Setup a string variable
	  set_fact:
	    my_name: "Max"
	 
	- name: Append string to list

	  set_fact:
	    my_strings: "{{ my_strings + [ my_name ] }}"

	- debug: var=my_strings

	- name: Append another item to the list
	  set_fact:
	    my_strings: "{{ my_strings + [ 'Power' ] }}"

	- debug: var=my_strings
```

**Comment by davestephens** [comment]
I'm not sure your assumptions are correct. Traefik only requests *one* certificate, and all of the various subdomains are requested as [SANs](https://knowledge.digicert.com/solution/SO9440.html) - so there's no cost in having many subdomains.

That said, there is an open PR (#44) to initially request the LetsEncrypt cert as a staging cert rather than production - it just needs some work before I'm happy to merge it.

**Comment by animeai** [comment]
Ahhh, I've misread the code then!

Looking at #44 I can probably use that to make the subdomains editable through variables and only request the active containers too. I was unaware that I could pass variables from ansible straight into the traefik.toml like that. Shows how much I have to learn about the software still!

**Comment by davestephens** [comment]
I'm not sure it needs changing from how it currently works, unless I'm
missing something?

Right now one cert is requested when Traefik starts - if you enable/disable
containers no new cert is required.

With what you're proposing a new cert would be requested every time you
enable/disable something, which could happen many times over a short period
of time - ending in you getting rate limited.

The only time a new cert is requested is if the existing cert is close to
expiry, or if a PR is merged with a new container that adds a new hostname
to the list of SANs.

**Comment by animeai** [comment]
Ahh I see what you mean actually. What I'm also proposing though is the ability to edit the subdomain names through all.yml variables alongside all of the other potential container variables one might wish to edit (restart policy, image, memory limit etc). It would make the package more flexible.

Is it just the documentation in #44 that needed work still?

**Comment by davestephens** [comment]
It also needs some work mentioned at the bottom of the issue, something @C-J1 and I discussed but we never documented :)

Being able to change the hostnames would be nice I guess, in case someone doesn't like the ones I/contributors picked. That does come with the downside you're more likely to hit LetsEncrypt rate limits though...

**Comment by animeai** [comment]
Fixed with #178

**Comment by davestephens** [comment]
Closing as this is now supported with the move to wildcard certs and ansible roles 👍
